components/UnifiedDocFeed/utils/getSelectedUrlFilters.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  feedTypeOpts,
+  sortOpts,
+  tagFilters,
+  topLevelFilters,
+} from "../constants/UnifiedDocFilters";
+import { scopeOptions } from "~/config/utils/options";
+
+export type SelectedUrlFilters = {
+  topLevel: string | undefined;
+  type: string | undefined;
+  sort: string | undefined;
+  time: string | undefined;
+  tags: string[];
+};
+
+export const getSelectedUrlFilters = ({
+  query,
+  pathname,
+}): SelectedUrlFilters => {
+  const defaults = {
+    topLevel: topLevelFilters[0].value,
+    type: Object.values(feedTypeOpts)[0].value,
+    sort: sortOpts[0].value,
+    time: scopeOptions[0].value,
+    tags: <string[]>[],
+  };
+  const selected = { ...defaults };
+
+  if (!(query || pathname)) {
+    return defaults;
+  }
+
+  if (Array.isArray(query.tags)) {
+    selected.tags = [...query.tags];
+  } else if (query.tags) {
+    selected.tags.push(query.tags);
+  }
+
+  const foundSort = sortOpts.find((opt) => opt.value === query?.sort)?.value;
+  const foundTopLevelFilter = topLevelFilters.find(
+    (f) => f.url === pathname
+  )?.value;
+  const foundTypeFilter = Object.values(feedTypeOpts).find(
+    (opt) => opt.value === query?.type
+  )?.value;
+  const foundTimeScope = scopeOptions.find(
+    (opt) => opt.value === query?.time
+  )?.value;
```

Comment:
This segment is kinda very ugly. 
Do we really need to sort & find what's there within this component rather than having something passed down?

Reply:
The goal of this function is to generate an object from URL state. If we pass down things to it then it prevents the logic from being centralized.

There is something I could potentially though which is to update the structure I store config in `UnifiedFeedFilters`. I'll see if that's feasible